codec/src/main/java/io/netty5/handler/codec/DelimiterBasedFrameDecoder.java

```diff
@@ -288,12 +239,26 @@ protected Object decode0(ChannelHandlerContext ctx, ByteBuf buffer) throws Excep
             } else {
                 // Still discarding the buffer since a delimiter is not found.
                 tooLongFrameLength += buffer.readableBytes();
-                buffer.skipBytes(buffer.readableBytes());
+                buffer.skipReadable(buffer.readableBytes());
             }
             return null;
         }
     }
 
+    @Override
+    protected void handlerRemoved0(ChannelHandlerContext ctx) throws Exception {
+        if (delimiters != null) {
+            closeDelimiters(delimiters);
+        }
+        super.handlerRemoved0(ctx);
+    }
+
+    private static void closeDelimiters(Buffer[] delimiters) {
+        for (Buffer delimiter : delimiters) {
+            delimiter.close();
```

Comment:
should we ensure we also call `close` for all if one of them throws ?

Reply:
Do we want to ignore those exceptions or we want to throw them after the attempt to close all buffers?

Reply:
I would say throw

Reply:
@normanmaurer PTAL